[Feature:TAGrading] new 'grade inquiry only' button in grading index page

yanliw123 opened the pull request:
### What is the current behavior?
Given a large course, grading index page may have many grade inquiries, and it is burdensome to scroll and check each individual grade inquiries (although they are displayed with red box).

### What is the new behavior?
new 'grade inquiry only' button allow graders only view homework with grade inquiry
![image](https://github.com/Submitty/Submitty/assets/114595115/69018b5c-958b-4bb9-9b5d-be67b3c5cacc)

### Other information?
<!-- Is this a breaking change? -->
<!-- How did you test -->

ziesski commented:
Please check for PR Title Check and js-lint as it is failing the test, will review it tho :)

ziesski reviewed (approved):
Revoke my approval, sorry for the confusion, please take a look at my below change request.

ziesski reviewed (dismissed):
The next/prev button was not sticky with the grade inquiry only button. It would be more intuitive to place the button below "Enable Anonymous Mode". It needs on/off button (ex. Grade Inquiry Only: On, Grade Inquiry Only: off) so once you turn on, it will also have to change the next/prev button inside the grading panel to be Prev/Next Grade Inquiry. And once it is off, it will return to original state. 
![Screen Shot 2023-07-21 at 3 34 49 PM](https://github.com/Submitty/Submitty/assets/123261952/d20f5461-9a23-4ef6-bbb0-85b6f3495579)

ziesski commented on site/app/templates/grading/electronic/Details.twig:
```diff
@@ -446,7 +460,7 @@
 
     {% elseif graded_gradeable.getAutoGradedGradeable().getActiveVersion() == 0 %}
-        {% set contents = "Cancelled Submission" %}
+        {% set contents = "Canceasdlled Submission" %}
```
is this typo? please fix :)

ziesski commented:
I have tested the functionality / haven't don any code reviews yet.
The stickiness of grade inquiry is iffy.
![Screen Shot 2023-07-28 at 2 40 38 PM](https://github.com/Submitty/Submitty/assets/123261952/33ae3242-67d9-4b63-a2de-a3448396e436)
![Screen Shot 2023-07-28 at 2 40 48 PM](https://github.com/Submitty/Submitty/assets/123261952/d3e78a8c-fb2c-4ac0-8612-c0c75739d292)

While I am grading (grade inquiry only: off) the I tried to move to next student, but it kicks me out of the panel saying reached the end of list with inquiry filter. I double checked the setting, and it is set to next/prev student. 

Please check if your code is setting the filter in two different places.

ziesski commented:
2 Things I noticed during review.

First, I made the gradeable and first time I entered the Grading index page, the Grade Inquiry Only button was On, could we default to off? Instructor or TA should turn it on if they need the feature. 

Second, the button is sticky, I do see that when Grade inquiry only button is On the 'Prev/Next student arrow functionality' changes to Prev/Next Grade Inquiry and once the button is Off it switches back to original setting. However, it only changes the visible text, not the functionality. So although the setting is set to Prev/Next Grade inquiry, the arrow button will still be just 'Prev/Next Student'. In below screenshot, I turned the Grade inquiry only button On, but the right arrow will direct me to next student, instead of next inquiry student.

![Screen Shot 2023-08-01 at 4 47 20 PM](https://github.com/Submitty/Submitty/assets/123261952/c8ce284e-bf21-4d8a-8a18-88122c874ab5)

You may be only changing the superficial parts of code, please test the functionality by creating multiple inquiries from different students to ensure it is working as expected.

ziesski reviewed (changes requested):
2 Things I noticed during review.

First, I made the gradeable and first time I entered the Grading index page, the Grade Inquiry Only button was On, could we default to off? Instructor or TA should turn it on if they need the feature. 

Second, the button is sticky, I do see that when Grade inquiry only button is On the 'Prev/Next student arrow functionality' changes to Prev/Next Grade Inquiry and once the button is Off it switches back to original setting. However, it only changes the visible text, not the functionality. So although the setting is set to Prev/Next Grade inquiry, the arrow button will still be just 'Prev/Next Student'. In below screenshot, I turned the Grade inquiry only button On, but the right arrow will direct me to next student, instead of next inquiry student. The arrow buttons should work according to the setting. 

![Screen Shot 2023-08-01 at 4 47 20 PM](https://github.com/Submitty/Submitty/assets/123261952/c8ce284e-bf21-4d8a-8a18-88122c874ab5)

You may be only changing the superficial parts of code, please test the functionality by creating multiple inquiries from different students to ensure it is working as expected.

NiaHeermance reviewed (changes requested):
The progressing between students with inquiries works for me! It does traverse through students who don't have active grade inquiries so long as they had a grade inquiry but it was resolved. I think this probably should be changed so that it skips students whose grade inquiry is resolved., especially since the details page only shows unresolved inquires with the new filter.

yanliw123 commented:
> The progressing between students with inquiries works for me! It does traverse through students who don't have active grade inquiries so long as they had a grade inquiry but it was resolved. I think this probably should be changed so that it skips students whose grade inquiry is resolved., especially since the details page only shows unresolved inquires with the new filter.

I changed the sticky feature from "inquiry" to "active-inquiry". Now if an inquiry is closed. It will be skipped.

ziesski reviewed (changes requested):
The functionality is working as expected, I guess one last fix is reverse compatibility. When we switch back from Prev/Next Active inquiry to Prev/Next student setting inside panel, the index page should turn the Grade inquiry only: off, which will make it fully sticky. 

I will review the code more in detail but the functionality seems almost ready.

ziesski commented on site/app/templates/grading/electronic/Details.twig:
```diff
@@ -50,9 +50,29 @@
                 </button>
                 {% endif %}
+                {% if grade_inquiry_only_button %}
+                <button id="inquiryButton" class="btn btn-default" onclick="change_inquiry()">
```
Lets just switch the order to class, id, onclick. Maintaining same style as above codes :)

yanliw123 commented:
> The functionality is working as expected, I guess one last fix is reverse compatibility. When we switch back from Prev/Next Active inquiry to Prev/Next student setting inside panel, the index page should turn the Grade inquiry only: off, which will make it fully sticky.
> 
> I will review the code more in detail but the functionality seems almost ready.

I fixed the bug you mentioned and now it's fully sticky. I also added a new feature. Now the expected behavior is 

- 1. The setting will change accordingly based on the inquiry_only button
- 2. If the previous setting is "Pre/Next Itempool student" (for example). and you change the grade inquiry to 'on'. the setting will be "Pre/Next active-inquiry". Then if you change it back to 'off'. The setting will be "Pre/Next Itempool student" instead of the default. (sticky to the previous setting).
- 3. If you change the setting to "Pre/Next active-inquiry" manually and go back to the grading index page, the grade inquiry button will be 'on' and if you set the grade inquiry button to 'off'. And click on one homework, the grading setting will be changed to default ("Pre/Next Student").

NiaHeermance reviewed (changes requested):
It now skips students without active inquiries, which is great! However, the mouseover text on the Next Student button is now stuck on "Next student with an active inquiry" even when I have Grade Inquiry: off.

yanliw123 commented:
> It now skips students without active inquiries, which is great! However, the mouseover text on the Next Student button is now stuck on "Next student with an active inquiry" even when I have Grade Inquiry: off.

Solved, This bug only happens when the user set the setting to active-inquiry manually and turns off the grade-inquiry only in the grading index page. At this time the tooltip data should be set back to default.

NiaHeermance reviewed (approved):
Works perfectly. I approve!
